### PR TITLE
algebra: adding tests for `matrix_bufAlloc()` function

### PR DESCRIPTION
**Pull request by PNieck**
JIRA: PP-71

## Description

This PR is adding tests for function `matrix_buAlloc`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here). `armv7a9-zynq7000-qemu`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: algebra/tests/matrix/basic_interactions.c [hunk 1]
```diff
@@ -161,3 +174,156 @@ TEST_GROUP_RUNNER(group_matrix_at)
...
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_overflow2)
+{
+	/* This checks if matrix_buAlloc is ready for potential overflow while calculating rows * cols  */
+	int rows = INT_MAX;
+	int cols = INT_MAX;
+
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, rows, cols));
+}
```
Comment (niewim19) [inline algebra/tests/matrix/basic_interactions.c:314]:
```suggestion
TEST(group_matrix_bufAlloc, matrix_bufAlloc_invalidArgs)
{
	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, 0, 0));
	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, 0, COLS));
	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, ROWS, 0));
	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, sqrt(UINT_MAX), sqrt(UINT_MAX)));
	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, UINT_MAX, UINT_MAX));
}
```

For small amount of arguments edge/corner cases this might be more readable

**Review by niewim19 — CHANGES_REQUESTED** [review]
Tests are looking good. Remarks:

 - There is TEAR_DOWN error where double free is possible in certain cases.
 - As there is small amount of corner cases we can group them into one test to avoid code bloating.

File: algebra/tests/matrix/basic_interactions.c [hunk 2]
```diff
@@ -161,3 +174,185 @@ TEST_GROUP_RUNNER(group_matrix_at)
...
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_structElems)
+{
+	dynMat.rows = 0;
+	dynMat.cols = 0;
+	dynMat.transposed = 1;
+
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_SUCC, matrix_bufAlloc(&dynMat, ROWS, COLS));
```
[the remark this answers is not in the document]
Comment (PNieck) [reply algebra/tests/matrix/basic_interactions.c:205]: No, because `matrix_bufAlloc` does not change the pointer do `.data` if function fails

File: algebra/tests/matrix/basic_interactions.c [hunk 3]
```diff
@@ -161,3 +174,185 @@ TEST_GROUP_RUNNER(group_matrix_at)
...
+TEST_SETUP(group_matrix_bufAlloc)
+{
+}
+
+
+TEST_TEAR_DOWN(group_matrix_bufAlloc)
+{
+	free(dynMat.data);
+	dynMat.data = NULL;
+}
```
[the remark this answers is not in the document]
Comment (PNieck) [reply algebra/tests/matrix/basic_interactions.c:196]: No, because `matrix_bufAlloc` does not change the pointer do `.data` if it fails - as I mentioned above. But I will rewrite this to make it look clearer.

**Review by damianloew — COMMENTED** [review]
We are allocating the same matrix in all test cases. I think we could free dynMat.data before/after test case. Above that, in my opinion allocating some bigger amount of matrices may be helpful.

**Comment by PNieck** [comment]
@damianloew We free `dynMat.data` after every test case

```c
TEST_TEAR_DOWN(group_matrix_bufAlloc)
{
	free(dynMat.data);
}
```

**Review by niewim19 — APPROVED** [review]
LGTM
